src/borg_import/main.py

```diff
@@ -1,17 +1,82 @@
 import argparse
+import logging
+import shutil
+import subprocess
+import sys
+from pathlib import Path
 
 from .rsnapshots import get_snapshots
 
+log = logging.getLogger(__name__)
+
+
+def borg_import(args, archive_name, path, timestamp=None):
+    borg_cmdline = ['borg', 'create']
+    if timestamp:
+        borg_cmdline += '--timestamp', timestamp.isoformat()
+    borg_cmdline.append(args.create_options)
+
+    repository = args.repository.resolve()
+    location = '{}::{}'.format(repository, archive_name)
+    borg_cmdline.append(location)
+    borg_cmdline.append('.')
+
+    log.debug('Borg command line: %r', borg_cmdline)
+    log.debug('Borg working directory: %s', path)
+    subprocess.check_call(borg_cmdline, cwd=str(path))
+
 
 def main():
+    if not shutil.which('borg'):
+        print('Borg does not seem to be installed. Please install Borg first.')
+        print('See instructions at https://borgbackup.readthedocs.io/en/stable/installation.html')
+        return 1
     parser = argparse.ArgumentParser()
-    parser.add_argument("--rsnapshot-root", help="Path to rsnapshot root directory")
+    parser.add_argument("--rsnapshot-root", help="Path to rsnapshot root directory", type=Path)
+    parser.add_argument("--create-options", "-o", help="Additional borg create options", default='')
+    parser.add_argument("repository", metavar="REPOSITORY", help="Borg repository", type=Path)
+
+    parser.set_defaults(log_level=logging.WARNING)
+    parser.add_argument("--debug", action='store_const', dest='log_level', const=logging.DEBUG)
+
     args = parser.parse_args()
+    logging.basicConfig(level=args.log_level)
 
     if args.rsnapshot_root:
+        import_path = args.rsnapshot_root / 'borg-import-dir'
+        import_journal = args.rsnapshot_root / 'borg-import-dir.snapshot'
+        if import_path.exists():
+            print('{} exists. Cannot continue.'.format(import_path))
+            return 1
         for rsnapshot in get_snapshots(args.rsnapshot_root):
-            print(rsnapshot)
+            timestamp = rsnapshot['timestamp'].replace(microsecond=0)
+            snapshot_original_path = rsnapshot['path']
+
+            print('Importing {} (timestamp {})'.format(rsnapshot['name'], timestamp))
+            log.debug('  Moving {} -> {}'.format(rsnapshot['path'], import_path))
+
+            with import_journal.open('w') as fd:
+                fd.write('Current snapshot: %s\n' % rsnapshot['name'])
+                fd.write('Original path: %s\n' % snapshot_original_path)
+
+            snapshot_original_path.rename(import_path)
+
+            try:
+                borg_import(args, rsnapshot['name'], import_path, timestamp=timestamp)
+            finally:
+                log.debug('  Moving {} -> {}'.format(import_path, rsnapshot['path']))
+                import_path.rename(snapshot_original_path)
+                import_journal.unlink()
```

Comment:
Probably want an option to not do that. Trading "doesn't fiddle with the import source" and "it's fukkken fast!"